### ISSUE TITLE
Pose fist kpt Error

### ISSUE THREAD

**Issue by Zigars** [ultralytics/ultralytics#17847]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I use yolo11 pose model train meter kpt for meter reader, find it error in my first kpt prediction,  I trid put the fist kpt in other pose, and it stell can not get the right prediction.
![image](https://github.com/user-attachments/assets/a05ba4b6-723a-4dd3-be1b-10e47bc4dad6)
labels.jpg
![image](https://github.com/user-attachments/assets/cc08c641-980f-4b8d-bc24-a987347f47ae)
pred.jpg

### Environment

Ultralytics 8.3.36 🚀 Python-3.8.8 torch-1.8.1 CUDA:0 (NVIDIA GeForce RTX 4090, 24210MiB)
YOLO11n-pose summary (fused): 257 layers, 2,866,468 parameters, 0 gradients, 7.4 GFLOPs

### Minimal Reproducible Example

yolo cfg=config/cfg/JCQ/JCQ-POINT-20241126_V1_n_640_300_256.yaml

### Additional

no additional

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Zigars, thank you for reaching out and for your detailed report 🚀! We appreciate your interest in Ultralytics.

To assist you effectively, we kindly request a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This would include:

1. A concise script or command that we can run to reproduce the issue.
2. Any necessary supporting files like datasets, configuration files, or model weights. If these are too large, consider sharing a sample subset or anonymized example.
3. Additional information about your dataset setup and label format may also help us diagnose the problem.

For users new to Ultralytics, we recommend checking the [Docs](https://docs.ultralytics.com) for general usage guidance, such as [Pose](https://docs.ultralytics.com/tasks/pose/) setups, and ensuring your project aligns with documented best practices.

## Upgrade

Please also confirm that you are using the latest version of the `ultralytics` package, as bug fixes and improvements are regularly added. You can upgrade by running:

```bash
pip install -U ultralytics
```

This ensures compatibility with recent updates and helps confirm if the issue exists in the latest release.

## Environments

We strongly recommend testing your setup in one of our pre-configured and up-to-date environments to rule out dependency-related issues:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **AWS**: See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- [**Docker**](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/): Use our verified Docker image for consistency. <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Additionally, check if your environment matches our [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), such as **Python>=3.8** and **PyTorch>=1.8**, to ensure compatibility.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These tests validate functionality across macOS, Windows, and Ubuntu environments.

💡 This is an automated response, but please rest assured that an Ultralytics engineer will review your issue soon. Additional context, such as logs or visual examples (like the predictions you've already attached), are invaluable in helping us identify the issue faster.

Thank you! 😊

**Comment by Zigars** [comment]
Solve the problem by expand the boxes. Because  i create the box by external rectangular boxes of all points, This will result in the inability to obtain the correct gt when taking the kpt mask，This further leads to the inability to perform keypoint regression normally. 
So when creating a dataset, be careful not to take boxes directly from all kpt bounding boxes, and remember to further expand them

**Comment by Y-T-G** [comment]
You need to increase the box size to cover the whole scale

**Comment by Zigars** [comment]
> You need to increase the box size to cover the whole scale

yes, i find this can solve my problem!
